### PR TITLE
Publish Linux packages to S3

### PR DESCRIPTION
**Pull request by webrouse**
Jira: https://keboola.atlassian.net/browse/COM-1096

### CODE REVIEW

File: scripts/pkgs/publish-to-s3.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      ALPINE_ARCH="armhf"
+      ;;
+    *)
+      echo "Unexpected ARCH='$ARCH'"
+      exit 2
+      ;;
+  esac
+
+  # Copy
+  export DST_DIR="${S3_MOUNTPOINT}/alpine/${ALPINE_ARCH}";
```
Comment (webrouse) [inline scripts/pkgs/publish-to-s3.sh:54]: Alpine repository vyzaduje takyto format adresara, ktory obsahuje architekturu.

**Review by webrouse — COMMENTED** [review]
Funguje to takto:
- Goreleaser podpise jednotlive balicky.
- S3 bucket sa pripoji cez `s3fs`.
- Balicky sa skopiruju na S3. `./scripts/pkgs/publish-to-s3.sh`
- Vytvori sa novy index balickov, tj. prejdu sa vsetky balicky v S3, `./scripts/pkgs/index.sh`.

**Comment by webrouse** [comment]
Ked bude ta public S3 cez HTTPS, tak pridam dokumentaciu a testy, ze sa dane balicky daju nainstalovat.

File: .github/actions/mount-s3/action.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    - name: Mount S3 bucket
+      shell: bash
+      run: |
+        sudo apt install s3fs
+        sudo mkdir /s3bucket
+        sudo chown -R $(id -u):$(id -g) /s3bucket
+        echo $AWS_ACCESS_KEY_ID:$AWS_SECRET_ACCESS_KEY > ${HOME}/.passwd-s3fs
+        chmod 600 ${HOME}/.passwd-s3fs
+        echo -e "\nuser_allow_other\n" | sudo tee -a /etc/fuse.conf
+        s3fs -o allow_other $AWS_BUCKET_NAME /s3bucket
```
[the remark this answers is not in the document]
Comment (kacurez) [reply .github/actions/mount-s3/action.yml:15]: je nutne to mat cez s3fs? napr na kbc-ui pouzivaju jednoducy `aws s3 sync` ktory umozni nastavit dalsie parametre ako napr cache atd..
https://github.com/keboola/kbc-ui/blob/f718c5edb1cf48a5b81e973a3b08263a28d96fc1/.github/workflows/master.yml#L45-L47

[the remark this answers is not in the document]
Comment (webrouse) [reply .github/actions/mount-s3/action.yml:15]: @kacurez no casom tam budu GB dat, ktore som nechcel zbytocne prenasat, hore-dole (teda hlavne dole, do buildu).
Pouzite nastroje potrebuju pri generovani indexu vsetky subory, aj predchadzajuce, ... ale necitaju ich cele.
Preto mi to prislo ako dobre riesenie.

[the remark this answers is not in the document]
Comment (webrouse) [reply .github/actions/mount-s3/action.yml:15]: cc @ujovlado

[the remark this answers is not in the document]
Comment (webrouse) [reply .github/actions/mount-s3/action.yml:15]: Potom je este alternativne riesenie, ... ktore pracuje priamo s S3, ale je `no longer maintained`.

https://github.com/krobertson/deb-s3

**Comment by kacurez** [comment]
pozriem sa nato zajtra

File: .github/actions/mount-s3/action.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ujovlado) [reply .github/actions/mount-s3/action.yml:15]: Chapem, ide tam myslim len o to, ci ten tool, ktory tam mas vie objektom nastavit aj public access pripadne cache.

[the remark this answers is not in the document]
Comment (webrouse) [reply .github/actions/mount-s3/action.yml:15]: Jj, malo by sa to dat v `s3fs` nastavit:
https://github.com/s3fs-fuse/s3fs-fuse/blob/master/test/sample_ahbe.conf

Ale nie som si isty ci to pkg. managers beru do uvahy, skor asi nie.
Myslim, ze vzdy stahuju zoznam balickov pri napr. `apt-get update`.

File: .goreleaser.yml [hunk 3]
```diff
@@ -21,19 +21,20 @@ changelog:
...
 archives:
-  - files:
+  - id: archives
+    files:
       - none*
     format: zip
     format_overrides:
       - goos: darwin
         format: binary
-    name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
```
[the remark this answers is not in the document]
Comment (kacurez) [reply .goreleaser.yml:30]: potom sa bude musiet zmenit aj `release-mac-os` pretoze ten pocita stym namingom
https://github.com/keboola/keboola-as-code/blob/ff23444fbb0dd468733f99c3acdd5528f7b07bc8/.github/workflows/push.yml#L187
https://github.com/keboola/keboola-as-code/blob/ff23444fbb0dd468733f99c3acdd5528f7b07bc8/.github/workflows/push.yml#L164
rovnako aj `release-homebrew`
https://github.com/keboola/keboola-as-code/blob/ff23444fbb0dd468733f99c3acdd5528f7b07bc8/.github/workflows/push.yml#L228-L232

[the remark this answers is not in the document]
Comment (webrouse) [reply .goreleaser.yml:30]: Ok fixnem ... ono jediny rozdiel bude `linux_arm` ->` linux_armv6`

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -16,6 +16,21 @@ env:
   TEST_KBC_STORAGE_API_TOKEN: ${{ secrets.TEST_KBC_STORAGE_API_TOKEN }}
   TEST_KBC_STORAGE_API_TOKEN_MASTER: ${{ secrets.TEST_KBC_STORAGE_API_TOKEN_MASTER }}
   TEST_KBC_STORAGE_API_TOKEN_EXPIRED: ${{ secrets.TEST_KBC_STORAGE_API_TOKEN_EXPIRED }}
+
+  # S3 repository
+  IAM_USER: keboola-as-code-s3
+  AWS_BUCKET_NAME: keboola-as-code
```
Comment (kacurez) [inline .github/workflows/push.yml:22]: AWS_BUCKET_NAME je `keboola-as-code-dist`

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -16,6 +16,21 @@ env:
   TEST_KBC_STORAGE_API_TOKEN: ${{ secrets.TEST_KBC_STORAGE_API_TOKEN }}
   TEST_KBC_STORAGE_API_TOKEN_MASTER: ${{ secrets.TEST_KBC_STORAGE_API_TOKEN_MASTER }}
   TEST_KBC_STORAGE_API_TOKEN_EXPIRED: ${{ secrets.TEST_KBC_STORAGE_API_TOKEN_EXPIRED }}
+
+  # S3 repository
+  IAM_USER: keboola-as-code-s3
```
Comment (kacurez) [inline .github/workflows/push.yml:21]: iam user je `keboola-as-code-dist-s3-upload`

File: .github/workflows/push.yml (hunk 4 above)
Comment (webrouse) [reply .github/workflows/push.yml:22]: jj, mam pripraveny commit

File: .github/workflows/push.yml (hunk 5 above)
Comment (webrouse) [reply .github/workflows/push.yml:21]: ok, je to len, ak by ho niekto potreboval najst

**Comment by webrouse** [comment]
Vykusam doladim a dam ta opat na review, diky :beers:

**Review by webrouse — COMMENTED** [review]
- Presiel release s prazdnou S3: https://github.com/keboola/keboola-as-code/actions/runs/1217144246
- Aj ak tam uz bol nejaky release: https://github.com/keboola/keboola-as-code/actions/runs/1217375780
- Presiel aj opakovany release toho isteho tagu: 
- Takze by s tym nemali byt problemy.

**Review by kacurez — APPROVED** [review]
vidim ze cela pipeline presla tak super 👍

**Pull request merged**